src/fileio.cpp

```diff
@@ -481,6 +482,25 @@ FILE *FioFOpenFile(const char *filename, const char *mode, Subdirectory subdir,
 		strecpy(resolved_name, filename, lastof(resolved_name));
 		strtolower(resolved_name);
 
+		/* Resolve ".." */
+		std::stringstream ss(resolved_name);
+		std::vector<std::string> tokens;
+		for (std::string token; std::getline(ss, token, PATHSEPCHAR);) {
```

Comment:
Not really a fan of the for loop here, it seems a bit hacky. while loop might be nicer?